pkg/kubelet/pod_workers.go

```diff
@@ -981,25 +981,29 @@ func calculateEffectiveGracePeriod(status *podSyncStatus, pod *v1.Pod, options *
 	// enforce the restriction that a grace period can only decrease and track whatever our value is,
 	// then ensure a calculated value is passed down to lower levels
 	gracePeriod := status.gracePeriod
+
+	// make a best effort to default this value to the pod's desired intent, in the event
+    // the kubelet provided no requested value (graceful termination?)
+    if override := pod.Spec.TerminationGracePeriodSeconds; override != nil {
+        if *override < gracePeriod {
+            gracePeriod = *pod.Spec.TerminationGracePeriodSeconds
```

Comment:
can we not use
`gracePeriod = *override`
like the rest of the code blocks in this function?

Reply:
Thank you, done.

Reply:
Thank you for the fix. There is a current failure on the unit test regarding the mismatch of the pod's history: (TestUpdatePodDoesNotForgetSyncPodKill)  Was the unit test failing before implementing this fix according to this review comment?

Reply:
I was also able to confirm the behavior tests before this fix. It has been appearing before the fix above.

> Thank you for the fix. There is a current failure on the unit test regarding the mismatch of the pod's history: (TestUpdatePodDoesNotForgetSyncPodKill) Was the unit test failing before implementing this fix according to this review comment?

I will also see the cause of the test error, but I will resolve this thread of comments first.